/*
proto
*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package network;


service SNS{
    rpc Follow (FollowRequest) returns (FollowReply) {}
    rpc Unfollow (UnfollowRequest) returns (UnfollowReply) {}
    rpc List (ListRequest) returns (ListReply) {}
} //end SNS


message FollowRequest{
    // Client specifies name to follow
    string name = 1;
}

message FollowReply{
    int32 IReplyValue = 1;
}

message UnfollowRequest{
    // Client specifies name to unfollow
    string name = 1;
}

message UnfollowReply{
    int32 IReplyValue = 1;
}

message ListRequest{
    // This can be empty
    
}

message ListReply{
    repeated string names = 1;
}

